refgenconf/refgenconf.py

```diff
@@ -2087,11 +2164,148 @@ def _select_genomes(self, genome=None, strict=False, order=None,
+                            asset_path = rgc.seek(
+                                genome, "fasta", tag, enclosing_dir=True)
+                            ssc = SeqColClient({})
+                            digest, _ = ssc.load_fasta(
+                                os.path.join(asset_path, genome+'.fa'))
+                        except:
+                            continue
+                    continue
+
+            if digest:
+                # create "aliases" section
+                genome_v[CFG_ALIASES_KEY] = [genome]
+                # convert seek keys, childran/parent asset keys from aliases to genome digests
+                for asset, asset_v in genome_v[CFG_ASSETS_KEY].items():
+                    for tag, tag_v in asset_v[CFG_ASSET_TAGS_KEY].items():
+                        for seek, seek_v in tag_v[CFG_SEEK_KEYS_KEY].items():
+                            asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_SEEK_KEYS_KEY][seek] = seek_v.replace(
+                                genome, digest)
+
+                        if (asset == "fasta" and CFG_ASSET_CHILDREN_KEY in asset_v[CFG_ASSET_TAGS_KEY][tag]):
+                            for i in range(len(asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_CHILDREN_KEY])):
+                                asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_CHILDREN_KEY][i] = asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_CHILDREN_KEY][i].replace(
+                                    genome, digest)
+                        elif asset != "fasta":
+                            for i in range(len(asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_PARENTS_KEY])):
+                                asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_PARENTS_KEY][i] = asset_v[CFG_ASSET_TAGS_KEY][tag][CFG_ASSET_PARENTS_KEY][i].replace(
+                                    genome, digest)
+
+                # use the genome digest as primary keys
+                rgc[CFG_GENOMES_KEY][digest] = rgc[CFG_GENOMES_KEY].pop(
+                    genome)
+                # remove old "genome_digest" section
+                del rgc[CFG_GENOMES_KEY][digest][CFG_CHECKSUM_KEY]
+            else:
+                missing_digest.append(genome)
+                del rgc[CFG_GENOMES_KEY][genome]
+
+        return missing_digest
+
+    # restructure the genome_folder
+
+    def alter_file_tree():
+        my_genome = {}
+        for k, v in rgc[CFG_GENOMES_KEY].items():
+            my_genome.update([(v[CFG_ALIASES_KEY][0], k)])
+
+        os.mkdir(os.path.abspath(os.path.join(rgc[CFG_FOLDER_KEY], DATA_DIR)))
+        os.mkdir(os.path.abspath(os.path.join(rgc[CFG_FOLDER_KEY], ALIAS_DIR)))
+
+        # copy folder
+        for root, dirs, files in os.walk(rgc[CFG_FOLDER_KEY]):
+            for dir in dirs:
+                if dir in my_genome:
+                    shutil.copytree(os.path.join(rgc[CFG_FOLDER_KEY], dir),
+                                    os.path.join(rgc[CFG_FOLDER_KEY], DATA_DIR, dir), ignore=ignore_patterns("*.DS_Store"))
```

Comment:
don't bother with the `*.DS_Store` files, these are macOS-specific and there are probably hundreds other files that could be produced and we just don't care about them.. I think that users should take care of that